diff --git a/dist/index.js b/dist/index.js
index 87af5a8a60b4c89eeef75e1480b54f985a06e577..bac83206052a7def3a588a352a32197a95e1e686 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -5248,7 +5248,8 @@ var ethGetTransactionReceiptHandler = (client) => async (params) => {
       message: "No tx found"
     };
   }
-  const effectiveGasPrice = (
+  
+  const effectiveGasPrice = tx.maxFeePerGas ? (
     /** @type any*/
     tx.maxPriorityFeePerGas < /** @type any*/
     tx.maxFeePerGas - (block.header.baseFeePerGas ?? 0n) ? (
@@ -5258,7 +5259,8 @@ var ethGetTransactionReceiptHandler = (client) => async (params) => {
       /** @type any*/
       tx.maxFeePerGas - (block.header.baseFeePerGas ?? 0n) + (block.header.baseFeePerGas ?? 0n)
     )
-  );
+  ) : tx.gasPrice;
+
   vm.common.ethjsCommon.setHardfork(tx.common.hardfork());
   await vm.stateManager.setStateRoot(parentBlock.header.stateRoot);
   const runBlockResult = await vm.runBlock({
